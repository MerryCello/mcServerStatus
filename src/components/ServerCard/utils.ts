import arrowDown from '../../images/arrow-down.png';
import arrowUp from '../../images/arrow-up.png';
import arrowRight from '../../images/arrow-right.png';
import arrowDownSelected from '../../images/arrow-down-selected.png';
import arrowUpSelected from '../../images/arrow-up-selected.png';
import arrowRightSelected from '../../images/arrow-right-selected.png';

export const ARROWS = {
  'up-button': arrowUp,
  'down-button': arrowDown,
  'play-button': arrowRight,
};
export const ARROWS_SELECTED = {
  'up-button': arrowUpSelected,
  'down-button': arrowDownSelected,
  'play-button': arrowRightSelected,
};

export const getAttribute = (
  e: any,
  attr: string,
): keyof typeof ARROWS_SELECTED =>
  e.currentTarget.attributes.getNamedItem(attr)?.value;
