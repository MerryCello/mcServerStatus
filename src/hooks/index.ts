export * from "./useWindowDimensions";
export * from "./useButtonNavigate";
