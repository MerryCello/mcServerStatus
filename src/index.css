body {
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
    'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
    monospace;
}

@font-face {
  font-family: "Minecraft";
  src: local("Minecraft"),
    url("./fonts/Minecraftia.ttf") format("truetype");
  font-weight: normal;
}

@font-face {
  font-family: "Minecraft-bold";
  src: local("Minecraft-bold"),
    url("./fonts/Minecrafter_3.ttf") format("truetype");
  font-weight: bold;
}
