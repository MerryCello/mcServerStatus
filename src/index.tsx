import React from 'react';
import ReactDOM from 'react-dom/client';
import {createBrowserRouter, RouterProvider, Navigate} from 'react-router-dom';
import './index.css';
import App from './App';
import reportWebVitals from './reportWebVitals';
import NotFound404 from './pages/NotFound404';
import EditServerPage from './pages/EditServerPage';
import DeleteServerPage from './pages/DeleteServerPage';

const router = createBrowserRouter([
  {
    path: '*',
    element: <Navigate to='/mcServerStatus' replace />,
    errorElement: <NotFound404 />,
  },
  {
    path: "/mcServerStatus",
    element: <App />,
    errorElement: <NotFound404 />,
  },
  {
    path: '/mcServerStatus/edit',
    element: <EditServerPage />,
    errorElement: <NotFound404 />,
  },
  {
    path: '/edit',
    element: <Navigate to='/mcServerStatus/edit' replace />,
    errorElement: <NotFound404 />,
  },
  {
    path: '/mcServerStatus/add',
    element: <EditServerPage />,
    errorElement: <NotFound404 />,
  },
  {
    path: '/add',
    element: <Navigate to='/mcServerStatus/add' replace />,
    errorElement: <NotFound404 />,
  },
  {
    path: '/mcServerStatus/delete',
    element: <DeleteServerPage />,
    errorElement: <NotFound404 />,
  },
  {
    path: '/delete',
    element: <Navigate to='/mcServerStatus/delete' replace />,
    errorElement: <NotFound404 />,
  },
]);

ReactDOM.createRoot(document.getElementById('root') as HTMLElement).render(
  <React.StrictMode>
    <RouterProvider router={router} />
  </React.StrictMode>,
);

// If you want to start measuring performance in your app, pass a function
// to log results (for example: reportWebVitals(console.log))
// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
reportWebVitals();
