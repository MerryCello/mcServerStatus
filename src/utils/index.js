export const isNil = (obj) => obj === null || obj === undefined;
