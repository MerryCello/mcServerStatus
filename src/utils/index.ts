export const isNil = (obj: any): obj is null | undefined => obj === null || obj === undefined;
export const noop = () => { };
