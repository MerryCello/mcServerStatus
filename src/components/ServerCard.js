import { useState } from "react";
import parse from "html-react-parser";
import Signal from "./Signal";
import serverDefaultIcon from "../images/serverDefaultIcon.png";
import { useWindowDimensions } from "../hooks";

const ServerCard = ({
  name,
  address,
  status,
  style,
  onFocus,
  onBlur,
  onClick,
  tabIndex,
  index,
}) => {
  const [isSelected, setIsSelected] = useState(false);
  const icon = status?.icon;
  const motd = status?.motd;
  const playersOnline = status?.players?.online;
  const maxPlayers = status?.players?.max;

  const { width } = useWindowDimensions();
  const isMobileOrTablet = width < 500;

  const cardOnClick = () => {
    setIsSelected(true);
    if (onClick) {
      onClick(index);
    }
  };

  const cardOnFocus = () => {
    setIsSelected(true);
    if (onFocus) {
      onFocus(index);
    }
  };

  const cardOnBlur = (event) => {
    setIsSelected(false);
    if (onBlur) {
      onBlur(event);
    }
  };

  let i = 0;
  return (
    // TODO: put inline styles in CSS file
    <div
      className={"card-grid" + (isSelected ? " card-selected" : "")}
      style={style}
      onFocus={cardOnFocus}
      onBlur={cardOnBlur}
      // onFocus={cardOnClick}
      onClick={cardOnClick}
      tabIndex={tabIndex}
    >
      <div className="card-icon">
        <img src={icon ? icon : serverDefaultIcon} alt="icon for server" />
      </div>
      <div className="server-details">
        <div className="text-left">
          <div className="card-srv-name-status">
            <span title={address}>{name}</span>
            <span className="card-srv-name-status">
              {status?.online && (
                <span
                  className="pr-2 players-online"
                  style={{ color: "#7e7e7e" }}
                  title={
                    status?.players?.list
                      ? status.players.list.join("\n")
                      : status?.players?.online
                      ? "No players disclosed"
                      : "No players online"
                  }
                >
                  {status?.players
                    ? (playersOnline ?? "") + "/" + (maxPlayers ?? "")
                    : "-/-"}
                </span>
              )}
              <Signal
                server={{
                  hostname: address,
                  online: status?.online,
                  loading: status?.loading,
                  pingAvgMs: status?.pingAvgMs,
                }}
                size={1}
                style={{ paddingBottom: 5 }}
              />
            </span>
          </div>
        </div>
        <div className="text-left">
          <p
            className="m-0"
            style={
              status?.online && !status?.loading
                ? { color: "#7e7e7e" }
                : { color: "red" }
            }
          >
            {status?.online && motd
              ? parse(
                  motd?.html
                    ?.map((strHtml) => `<span key={${i++}}>${strHtml}</span>`)
                    .join("<br/>") || ""
                )
              : status?.loading
              ? ""
              : "Can't connect to server"}
          </p>
        </div>
      </div>
      {isMobileOrTablet && isSelected && (
        <>
          <p style={{ color: "#7e7e7e" }}>
            {parse(
              status?.version?.replace(/ /gm, "<br/>").replace(/,/gm, "") || ""
            )}
          </p>
          <p className="player-list">
            {status?.players?.list
              ? status.players.list.join(", ")
              : status?.players?.online
              ? "No players disclosed"
              : "No players online"}
          </p>
        </>
      )}
    </div>
  );
};

export default ServerCard;
