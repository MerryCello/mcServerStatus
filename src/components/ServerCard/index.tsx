import React, {CSSProperties, FC, FocusEventHandler, useState} from 'react';
import parse from 'html-react-parser';
import Signal from '../Signal';
import serverDefaultIcon from '../../images/serverDefaultIcon.png';
import {useWindowDimensions} from '../../hooks';
import {ServerStatus} from '../../types';

type ServerCardProps = {
  name: string;
  address: string;
  status: ServerStatus;
  style?: CSSProperties;
  onFocus: (index: number) => void;
  onBlur: FocusEventHandler<HTMLDivElement>;
  onClick?: (index: number) => void;
  tabIndex: number;
  index: number;
};

const ServerCard: FC<ServerCardProps> = ({
  name,
  address,
  status,
  style,
  onFocus,
  onBlur,
  onClick,
  tabIndex,
  index,
}) => {
  const [isSelected, setIsSelected] = useState(false);
  const icon = status?.icon;
  const motd = status?.motd;
  const playersOnline = status?.players?.online;
  const maxPlayers = status?.players?.max;

  const {width} = useWindowDimensions();
  const isMobileOrTablet = width < 500;

  const cardOnClick = () => {
    setIsSelected(true);
    onClick?.(index);
  };

  const cardOnFocus = () => {
    setIsSelected(true);
    onFocus?.(index);
  };

  const cardOnBlur: FocusEventHandler<HTMLDivElement> = (event) => {
    setIsSelected(false);
    onBlur?.(event);
  };

  return (
    // TODO: put inline styles in CSS file
    <div
      data-testid='server-card'
      className={'card-grid' + (isSelected ? ' card-selected' : '')}
      style={style}
      onFocus={cardOnFocus}
      onBlur={cardOnBlur}
      onClick={cardOnClick}
      tabIndex={tabIndex}>
      <div className='card-icon'>
        <img
          data-testid='server-icon'
          src={icon ? icon : serverDefaultIcon}
          alt='icon for server'
        />
      </div>
      <div className='server-details'>
        <div className='text-left'>
          <div className='card-srv-name-status'>
            <span data-testid='server-card-title' title={address}>
              {name}
            </span>
            <span className='card-srv-name-status'>
              {status?.online && (
                <span
                  data-testid='server-player-count'
                  className='pr-2 players-online'
                  style={{color: '#7e7e7e'}}
                  title={
                    status?.players?.list
                      ? status.players.list.join('\n')
                      : status?.players?.online
                      ? 'No players disclosed'
                      : 'No players online'
                  }>
                  {`${playersOnline ?? '-'}/${maxPlayers ?? '-'}`}
                </span>
              )}
              <Signal
                data-testid='server-signal-strength'
                server={{
                  hostname: address,
                  online: status?.online,
                  loading: status?.loading,
                  pingAvgMs: status?.pingAvgMs,
                }}
                size={1}
                style={{paddingBottom: 5}}
              />
            </span>
          </div>
        </div>
        <div className='text-left'>
          <p
            data-testid='server-card-body'
            className='m-0'
            style={status?.online ? {color: '#7e7e7e'} : {color: 'red'}}>
            {status?.online && motd
              ? parse(
                  motd?.html
                    ?.map(
                      (strHtml, i) =>
                        `<span data-testid="motd-html" key={${i}}>${strHtml}</span>`,
                    )
                    .join('<br/>') || '',
                )
              : status?.loading
              ? ''
              : "Can't connect to server"}
          </p>
        </div>
      </div>
      {isMobileOrTablet && isSelected && (
        <>
          <p
            data-testid='mobile-server-card-versions'
            style={{color: '#7e7e7e'}}>
            {parse(
              status?.version?.replace(/ /gm, '<br/>').replace(/,/gm, '') || '',
            )}
          </p>
          <p
            data-testid='mobile-server-card-player-list'
            className='player-list'>
            {status?.players?.list
              ? status.players.list.join(', ')
              : status?.players?.online
              ? 'No players disclosed'
              : 'No players online'}
          </p>
        </>
      )}
    </div>
  );
};

export default ServerCard;
