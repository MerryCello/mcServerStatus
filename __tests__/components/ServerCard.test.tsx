/** @jest-environment jsdom */
import React, {useState} from 'react';
import {render} from '@testing-library/react';
import ServerCard from '../../src/components/ServerCard';
import {noop} from '../../src/utils';
import {useWindowDimensions} from '../../src/hooks';

jest.mock('../../src/hooks/useWindowDimensions');
jest.mock('../../src/components/Signal', () => (props) => (
  // @ts-expect-error
  <signal {...props} server={JSON.stringify(props.server)} />
));

describe('ServerCard', () => {
  it('should render data state', () => {
    (useWindowDimensions as jest.Mock).mockReturnValue({
      width: 1000,
      height: 800,
    });
    const serverStatus = {
      online: true,
      loading: false,
      pingAvgMs: 60,
      icon: 'test-icon',
      motd: {
        raw: ['§l§9Worlds Without End§r§r'],
        clean: ['Worlds Without End'],
        html: [
          '<span style="color: #5555FF">Worlds Without End</span><br/><span>This Awesome!</span>',
        ],
      },
      players: {
        online: 1,
        max: 10,
        list: ['test-player-1', 'test-player-2', 'test-player-3'],
      },
    };
    const props = {
      name: 'Test Server',
      address: 'test.server.com',
      status: serverStatus,
      style: {width: 200, height: 200},
      onFocus: jest.fn(),
      onBlur: jest.fn(),
      tabIndex: 0,
      index: 0,
    };

    // @ts-ignore - This is a bad players.list type definition. Should be string[].
    const {container, getByTestId} = render(<ServerCard {...props} />);

    const card = getByTestId('server-card');
    expect(card).toBeInTheDocument();
    expect(card).toHaveAttribute('style', 'width: 200px; height: 200px;');
    expect(card).toHaveAttribute('tabindex', props.tabIndex.toString());
    // Icon
    const serverIcon = getByTestId('server-icon');
    expect(serverIcon).toBeInTheDocument();
    expect(serverIcon).toHaveAttribute('src', serverStatus.icon);
    expect(serverIcon).toHaveAttribute('alt');
    // Title
    const serverTitle = getByTestId('server-card-title');
    expect(serverTitle).toBeInTheDocument();
    expect(serverTitle).toHaveTextContent(props.name);
    expect(serverTitle).toHaveAttribute('title', props.address);
    // Player Count
    const serverPlayerCount = getByTestId('server-player-count');
    expect(serverPlayerCount).toBeInTheDocument();
    expect(serverPlayerCount).toHaveTextContent(
      `${serverStatus.players.online}/${serverStatus.players.max}`,
    );
    expect(serverPlayerCount).toHaveAttribute(
      'title',
      serverStatus.players.list.join('\n'),
    );
    // Signal Strength
    const signalStrength = getByTestId('server-signal-strength');
    expect(signalStrength).toBeInTheDocument();
    expect(signalStrength).toHaveAttribute(
      'server',
      JSON.stringify({
        hostname: props.address,
        online: serverStatus.online,
        loading: serverStatus.loading,
        pingAvgMs: serverStatus.pingAvgMs,
      }),
    );
    // Body
    const serverDetails = getByTestId('server-card-body');
    expect(serverDetails).toBeInTheDocument();
    expect(serverDetails).toContainElement(getByTestId('motd-html'));

    expect(container).toMatchSnapshot();
  });

  it('should render error/offline state', () => {
    const {container, getByTestId} = render(
      <ServerCard
        name='Test Server'
        address='test.server.com'
        status={{online: false}}
        onFocus={noop}
        onBlur={noop}
        tabIndex={0}
        index={0}
      />,
    );
    const serverDetails = getByTestId('server-card-body');
    expect(serverDetails).toBeInTheDocument();
    expect(serverDetails).toHaveTextContent("Can't connect to server");

    expect(container).toMatchSnapshot();
  });

  it('should render loading state', () => {
    const {container, queryByTestId, getByTestId} = render(
      <ServerCard
        name='Test Server'
        address='test.server.com'
        status={{loading: true}}
        onFocus={noop}
        onBlur={noop}
        tabIndex={0}
        index={0}
      />,
    );
    expect(queryByTestId('server-player-count')).toBeNull();
    const serverDetails = getByTestId('server-card-body');
    expect(serverDetails).toBeInTheDocument();
    expect(serverDetails).toHaveTextContent('');

    expect(container).toMatchSnapshot();
  });

  it('should call onFocus when focused', () => {
    const onFocus = jest.fn();
    const setIsSelected = jest.fn();
    (useState as jest.Mock).mockReturnValue([false, setIsSelected]);
    const serverCard = render(
      <ServerCard
        name='Test Server'
        address='test.server.com'
        status={{}}
        onFocus={onFocus}
        onBlur={noop}
        tabIndex={0}
        index={0}
      />,
    );
    const card = serverCard.getByTestId('server-card');
    card.focus();
    expect(onFocus).toHaveBeenCalledWith(0);
    expect(setIsSelected).toHaveBeenCalledWith(true);
  });

  it('should call onBlur when blurred', () => {
    const onBlur = jest.fn();
    const setIsSelected = jest.fn();
    (useState as jest.Mock).mockReturnValue([true, setIsSelected]);
    (useWindowDimensions as jest.Mock).mockReturnValueOnce({
      width: 450,
      height: 800,
    });
    const serverCardElement = () => (
      <ServerCard
        name='Test Server'
        address='test.server.com'
        // @ts-expect-error
        status={{online: true, motd: {}}}
        onFocus={noop}
        onBlur={onBlur}
        tabIndex={0}
        index={0}
      />
    );
    const serverCard = render(serverCardElement());
    const card = serverCard.getByTestId('server-card');
    card.focus();
    card.blur();
    expect(onBlur).toHaveBeenCalled();
    expect(setIsSelected).toHaveBeenCalledWith(false);
    (useState as jest.Mock).mockReturnValue([false, setIsSelected]);
    serverCard.rerender(serverCardElement());
    expect(card).not.toHaveClass('card-selected');
  });

  it('should call onClick with index when clicked', () => {
    const onClick = jest.fn();
    const setIsSelected = jest.fn();
    (useState as jest.Mock).mockReturnValue([false, setIsSelected]);
    const serverCardElement = () => (
      <ServerCard
        name='Test Server'
        address='test.server.com'
        status={{}}
        onFocus={noop}
        onBlur={noop}
        onClick={onClick}
        tabIndex={0}
        index={0}
      />
    );
    const {getByTestId, rerender} = render(serverCardElement());
    const card = getByTestId('server-card');
    expect(card).not.toHaveClass('card-selected');
    card.click();
    expect(onClick).toHaveBeenCalledWith(0);
    expect(setIsSelected).toHaveBeenCalledWith(true);
    (useState as jest.Mock).mockReturnValue([true, setIsSelected]);
    rerender(serverCardElement());
    expect(card).toHaveClass('card-selected');
  });

  it('should show "no players online" when no players are online', () => {
    const serverStatus = {
      online: true,
      motd: {html: ['A Server'], raw: [], clean: []},
      players: {
        online: 0,
        max: 10,
      },
    };
    const {container, getByTestId} = render(
      <ServerCard
        name='Test Server'
        address='test.server.com'
        status={serverStatus}
        onFocus={noop}
        onBlur={noop}
        tabIndex={0}
        index={0}
      />,
    );
    const serverPlayerCount = getByTestId('server-player-count');
    expect(serverPlayerCount).toBeInTheDocument();
    expect(serverPlayerCount).toHaveTextContent(
      `${serverStatus.players.online}/${serverStatus.players.max}`,
    );
    expect(serverPlayerCount).toHaveAttribute('title', 'No players online');
    expect(container).toMatchSnapshot();
  });

  it('should show "No players disclosed" when players are online but not disclosed', () => {
    (useWindowDimensions as jest.Mock).mockReturnValueOnce({
      width: 450,
      height: 800,
    });
    const serverStatus = {
      online: true,
      motd: {html: ['A Server'], raw: [], clean: []},
      players: {
        online: 5,
        max: 10,
      },
    };
    const {container, getByTestId} = render(
      <ServerCard
        name='Test Server'
        address='test.server.com'
        status={serverStatus}
        onFocus={noop}
        onBlur={noop}
        tabIndex={0}
        index={0}
      />,
    );
    const serverPlayerCount = getByTestId('server-player-count');
    expect(serverPlayerCount).toBeInTheDocument();
    expect(serverPlayerCount).toHaveTextContent(
      `${serverStatus.players.online}/${serverStatus.players.max}`,
    );
    expect(serverPlayerCount).toHaveAttribute('title', 'No players disclosed');
    expect(container).toMatchSnapshot();
  });

  it('should show player list for mobile view when selected', () => {
    (useState as jest.Mock).mockReturnValue([true, noop]);
    (useWindowDimensions as jest.Mock).mockReturnValueOnce({
      width: 450,
      height: 800,
    });
    const serverStatus = {
      online: true,
      motd: {raw: [], clean: [], html: ['Worlds Without End']},
      version: '1.16.5, 1.17.1',
      players: {
        online: 1,
        max: 10,
        list: ['test-player-1', 'test-player-2', 'test-player-3'],
      },
    };
    const {container, getByTestId} = render(
      <ServerCard
        name='Test Server'
        address='test.server.com'
        // @ts-ignore - This is a bad players.list type definition. Should be string[].
        status={serverStatus}
        onFocus={noop}
        onBlur={noop}
        tabIndex={0}
        index={0}
        isSelected={true}
      />,
    );
    const mobileCardVersions = getByTestId('mobile-server-card-versions');
    expect(mobileCardVersions).toBeInTheDocument();
    expect(mobileCardVersions).toHaveTextContent('1.16.51.17.1');
    const mobileCardPlayers = getByTestId('mobile-server-card-player-list');
    expect(mobileCardPlayers).toBeInTheDocument();
    expect(mobileCardPlayers).toHaveTextContent(
      serverStatus.players.list.join(', '),
    );

    expect(container).toMatchSnapshot();
  });
});
