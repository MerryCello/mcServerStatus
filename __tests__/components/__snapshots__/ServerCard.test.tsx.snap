// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServerCard should render data state 1`] = `
<div>
  <div
    class="card-grid"
    data-testid="server-card"
    style="width: 200px; height: 200px;"
    tabindex="0"
  >
    <div
      class="card-icon"
    >
      <img
        alt="icon for server"
        data-testid="server-icon"
        src="test-icon"
      />
    </div>
    <div
      class="server-details"
    >
      <div
        class="text-left"
      >
        <div
          class="card-srv-name-status"
        >
          <span
            data-testid="server-card-title"
            title="test.server.com"
          >
            Test Server
          </span>
          <span
            class="card-srv-name-status"
          >
            <span
              class="pr-2 players-online"
              data-testid="server-player-count"
              style="color: rgb(126, 126, 126);"
              title="test-player-1
test-player-2
test-player-3"
            >
              1/10
            </span>
            <signal
              data-testid="server-signal-strength"
              server="{\\"hostname\\":\\"test.server.com\\",\\"online\\":true,\\"loading\\":false,\\"pingAvgMs\\":60}"
              size="1"
              style="padding-bottom: 5px;"
            />
          </span>
        </div>
      </div>
      <div
        class="text-left"
      >
        <p
          class="m-0"
          data-testid="server-card-body"
          style="color: rgb(126, 126, 126);"
        >
          <span
            data-testid="motd-html"
          >
            <span
              style="color: rgb(85, 85, 255);"
            >
              Worlds Without End
            </span>
            <br />
            <span>
              This Awesome!
            </span>
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`ServerCard should render error/offline state 1`] = `
<div>
  <div
    class="card-grid"
    data-testid="server-card"
    tabindex="0"
  >
    <div
      class="card-icon"
    >
      <img
        alt="icon for server"
        data-testid="server-icon"
        src="test-file-stub"
      />
    </div>
    <div
      class="server-details"
    >
      <div
        class="text-left"
      >
        <div
          class="card-srv-name-status"
        >
          <span
            data-testid="server-card-title"
            title="test.server.com"
          >
            Test Server
          </span>
          <span
            class="card-srv-name-status"
          >
            <signal
              data-testid="server-signal-strength"
              server="{\\"hostname\\":\\"test.server.com\\",\\"online\\":false}"
              size="1"
              style="padding-bottom: 5px;"
            />
          </span>
        </div>
      </div>
      <div
        class="text-left"
      >
        <p
          class="m-0"
          data-testid="server-card-body"
          style="color: red;"
        >
          Can't connect to server
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`ServerCard should render loading state 1`] = `
<div>
  <div
    class="card-grid"
    data-testid="server-card"
    tabindex="0"
  >
    <div
      class="card-icon"
    >
      <img
        alt="icon for server"
        data-testid="server-icon"
        src="test-file-stub"
      />
    </div>
    <div
      class="server-details"
    >
      <div
        class="text-left"
      >
        <div
          class="card-srv-name-status"
        >
          <span
            data-testid="server-card-title"
            title="test.server.com"
          >
            Test Server
          </span>
          <span
            class="card-srv-name-status"
          >
            <signal
              data-testid="server-signal-strength"
              server="{\\"hostname\\":\\"test.server.com\\",\\"loading\\":true}"
              size="1"
              style="padding-bottom: 5px;"
            />
          </span>
        </div>
      </div>
      <div
        class="text-left"
      >
        <p
          class="m-0"
          data-testid="server-card-body"
          style="color: red;"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`ServerCard should show "No players disclosed" when players are online but not disclosed 1`] = `
<div>
  <div
    class="card-grid card-selected"
    data-testid="server-card"
    tabindex="0"
  >
    <div
      class="card-icon"
    >
      <img
        alt="icon for server"
        data-testid="server-icon"
        src="test-file-stub"
      />
    </div>
    <div
      class="server-details"
    >
      <div
        class="text-left"
      >
        <div
          class="card-srv-name-status"
        >
          <span
            data-testid="server-card-title"
            title="test.server.com"
          >
            Test Server
          </span>
          <span
            class="card-srv-name-status"
          >
            <span
              class="pr-2 players-online"
              data-testid="server-player-count"
              style="color: rgb(126, 126, 126);"
              title="No players disclosed"
            >
              5/10
            </span>
            <signal
              data-testid="server-signal-strength"
              server="{\\"hostname\\":\\"test.server.com\\",\\"online\\":true}"
              size="1"
              style="padding-bottom: 5px;"
            />
          </span>
        </div>
      </div>
      <div
        class="text-left"
      >
        <p
          class="m-0"
          data-testid="server-card-body"
          style="color: rgb(126, 126, 126);"
        >
          <span
            data-testid="motd-html"
          >
            A Server
          </span>
        </p>
      </div>
    </div>
    <p
      data-testid="mobile-server-card-versions"
      style="color: rgb(126, 126, 126);"
    />
    <p
      class="player-list"
      data-testid="mobile-server-card-player-list"
    >
      No players disclosed
    </p>
  </div>
</div>
`;

exports[`ServerCard should show "no players online" when no players are online 1`] = `
<div>
  <div
    class="card-grid card-selected"
    data-testid="server-card"
    tabindex="0"
  >
    <div
      class="card-icon"
    >
      <img
        alt="icon for server"
        data-testid="server-icon"
        src="test-file-stub"
      />
    </div>
    <div
      class="server-details"
    >
      <div
        class="text-left"
      >
        <div
          class="card-srv-name-status"
        >
          <span
            data-testid="server-card-title"
            title="test.server.com"
          >
            Test Server
          </span>
          <span
            class="card-srv-name-status"
          >
            <span
              class="pr-2 players-online"
              data-testid="server-player-count"
              style="color: rgb(126, 126, 126);"
              title="No players online"
            >
              0/10
            </span>
            <signal
              data-testid="server-signal-strength"
              server="{\\"hostname\\":\\"test.server.com\\",\\"online\\":true}"
              size="1"
              style="padding-bottom: 5px;"
            />
          </span>
        </div>
      </div>
      <div
        class="text-left"
      >
        <p
          class="m-0"
          data-testid="server-card-body"
          style="color: rgb(126, 126, 126);"
        >
          <span
            data-testid="motd-html"
          >
            A Server
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`ServerCard should show player list for mobile view when selected 1`] = `
<div>
  <div
    class="card-grid card-selected"
    data-testid="server-card"
    tabindex="0"
  >
    <div
      class="card-icon"
    >
      <img
        alt="icon for server"
        data-testid="server-icon"
        src="test-file-stub"
      />
    </div>
    <div
      class="server-details"
    >
      <div
        class="text-left"
      >
        <div
          class="card-srv-name-status"
        >
          <span
            data-testid="server-card-title"
            title="test.server.com"
          >
            Test Server
          </span>
          <span
            class="card-srv-name-status"
          >
            <span
              class="pr-2 players-online"
              data-testid="server-player-count"
              style="color: rgb(126, 126, 126);"
              title="test-player-1
test-player-2
test-player-3"
            >
              1/10
            </span>
            <signal
              data-testid="server-signal-strength"
              server="{\\"hostname\\":\\"test.server.com\\",\\"online\\":true}"
              size="1"
              style="padding-bottom: 5px;"
            />
          </span>
        </div>
      </div>
      <div
        class="text-left"
      >
        <p
          class="m-0"
          data-testid="server-card-body"
          style="color: rgb(126, 126, 126);"
        >
          <span
            data-testid="motd-html"
          >
            Worlds Without End
          </span>
        </p>
      </div>
    </div>
    <p
      data-testid="mobile-server-card-versions"
      style="color: rgb(126, 126, 126);"
    >
      1.16.5
      <br />
      1.17.1
    </p>
    <p
      class="player-list"
      data-testid="mobile-server-card-player-list"
    >
      test-player-1, test-player-2, test-player-3
    </p>
  </div>
</div>
`;
